### PR TITLE
Feat: IGM

### PR DESCRIPTION
**Pull request by sharmasahil0331**

### CODE REVIEW

File: webserver/main/service/issue_status_service.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    on_issue_status_message_id = message['message_ids']['on_issue_status']
+    on_issue_status_payload = get_first_ondc_request(
+        OndcDomain.RETAIL, OndcAction('on_issue_status'), on_issue_status_message_id)
+    bap_endpoint = on_issue_status_payload['context']['bap_uri']
+    status_code = make_request_over_ondc_network(
+        on_issue_status_payload, bap_endpoint, 'on_issue_status')
+    log(f"Sent responses to bg/bap with status-code {status_code}")
+
+
+if __name__ == "__main__":
```
Comment (navdeep710) [inline webserver/main/service/issue_status_service.py:54]: remove this, seems redundant

File: webserver/requirements.txt [hunk 2]
```diff
@@ -30,6 +30,6 @@ pymongo==4.1.1
 ed25519==1.5
 PyNaCl==1.5.0
 pika==1.3.0
-psycopg2-binary==2.9.1
+psycopg2==2.9.3
```
Comment (navdeep710) [inline webserver/requirements.txt:33]: hi this does not work with most systems, linux and mac requires binary, if you want you use separate env

Comment (sharmasahil0331) [reply webserver/requirements.txt:33]: Alright, i am reverting.

File: webserver/consumer_dockerfile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 FROM ubuntu:18.04
 ENV DEBIAN_FRONTEND noninteractive
 
-RUN apt-get update --fix-missing && apt-get install -y  python3.7 python3.7-dev python3-pip python3.7-venv supervisor virtualenv
+RUN apt-get update --fix-missing && apt-get install -y  python3.7 python3.7-dev python3-pip python3.7-venv supervisor virtualenv libpq-dev cargo build-essential libssl-dev libffi-dev
```
[the remark this answers is not in the document]
Comment (sharmasahil0331) [reply webserver/consumer_dockerfile:4]: cryptography and postgres

File: webserver/main/service/issue_status_service.py (hunk 1 above)
Comment (sharmasahil0331) [reply webserver/main/service/issue_status_service.py:54]: We have this log to monitor the status code of the request, as we are having in the other services.

Comment (navdeep710) [reply webserver/main/service/issue_status_service.py:54]: hi, sorry for the confusion, I was mentioning the main entry check you have put

File: webserver/consumer_dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (navdeep710) [reply webserver/consumer_dockerfile:4]: postgres is already present i guess 🤔

[the remark this answers is not in the document]
Comment (sharmasahil0331) [reply webserver/consumer_dockerfile:4]: But it was throwing errors on our local, should I revert those changes ?

File: webserver/main/service/issue_service.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    on_issue_payload = get_first_ondc_request(
+        OndcDomain.RETAIL, OndcAction('on_issue'), on_issue_message_id)
+    bap_endpoint = on_issue_payload['context']['bap_uri']
+    status_code = make_request_over_ondc_network(
+        on_issue_payload, bap_endpoint, 'on_issue')
+    log(f"Sent responses to bg/bap with status-code {status_code}")
+
+
+if __name__ == "__main__":
+    pass
```
Comment (navdeep710) [inline webserver/main/service/issue_service.py:55]: can we remove these statements ?

Comment (sharmasahil0331) [reply webserver/main/service/issue_service.py]: Removed

**Review by navdeep710 — APPROVED** [review]

**Pull request merged**